File: school-management-frontend/src/services/classService.js
import api from '../api/axios';

const classService = {
  getAllClasses: () => api.get('/classes'),
  getAllClassesWithCount: () => api.get('/classes/with-count'),
  getClassById: (id) => api.get(`/classes/${id}`),
  createClass: (classData) => api.post('/classes', classData),
  updateClass: (id, classData) => api.put(`/classes/${id}`, classData),
  deleteClass: (id) => api.delete(`/classes/${id}`),
};

export default classService;

File: school-management-backend/src/main/java/com/school/management/service/ClassService.java
package com.school.management.service;

import com.school.management.dto.ClassRequest;
import com.school.management.entity.SchoolClass;

import java.util.List;
import java.util.Map;

public interface ClassService {
    List<SchoolClass> getAllClasses();
    SchoolClass getClassById(Long id);
    SchoolClass createClass(ClassRequest request);
    SchoolClass updateClass(Long id, ClassRequest request);
    void deleteClass(Long id);
    List<Map<String, Object>> getAllClassesWithStudentCount();
}
